nemo/collections/nlp/models/language_modeling/megatron_base_prompt_learning_model.py

```diff
@@ -436,6 +436,12 @@ def save_checkpoint_as_nemo_file(self):
         self.virtual_prompt_style = current_virtual_prompt_style
         self.virtual_prompt_source = current_virtual_prompt_source
 
+        # Revert prompt table back to previous state
+        if self.virtual_prompt_style == VirtualPromptStyle.P_TUNING and self.first_stage_of_pipeline():
```

Comment:
Ideally, we would want to move any mentions of a specific prompt type outside of the base class. ok for now.